### PR TITLE
chore: update react and testing-library

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

As discussed, we update React and `testing-library` in a separate PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -161,7 +161,7 @@
     "jest-silent-reporter": "0.1.2",
     "jest-validate": "24.9.0",
     "jest-watch-master": "1.0.0",
-    "jest-watch-typeahead": "0.3.1",
+    "jest-watch-typeahead": "0.4.0",
```
Comment (montezume) [inline package.json]: drops node 6

File: test/setup-tests.js [hunk 2]
```diff
@@ -2,6 +2,11 @@ import fs from 'fs';
 import path from 'path';
 import colors from 'colors/safe';
 
+const shouldSilenceWarnings = (...messages) =>
+  [/Warning: componentWillReceiveProps has been renamed/].some(msgRegex =>
+    messages.some(msg => msgRegex.test(msg))
```
Comment (montezume) [inline test/setup-tests.js]: same approach as app kit

File: src/components/fields/localized-multiline-text-field/localized-multiline-text-field.spec.js [hunk 3]
```diff
@@ -23,16 +23,20 @@ class Story extends React.Component {
...
   };
+
   state = {
     value: this.props.value,
   };
+
   handleChange = event => {
+    const value = event.target.value;
     if (this.props.onChange) this.props.onChange(event);
+    event.persist();
```
Comment (tdeekens) [inline src/components/fields/localized-multiline-text-field/localized-multiline-text-field.spec.js]: Should this happen before invokving the onChange too?

Comment (montezume) [reply src/components/fields/localized-multiline-text-field/localized-multiline-text-field.spec.js]: moved in [80554ad22bbd8c658662270c0ba6462112734454](https://github.com/commercetools/ui-kit/pull/1016/commits/80554ad22bbd8c658662270c0ba6462112734454)

**Pull request merged**